samples/Sentry.Samples.Console.Basic/Sentry.Samples.Console.Basic.csproj

```diff
@@ -4,16 +4,10 @@
     <OutputType>Exe</OutputType>
     <ImplicitUsings>enable</ImplicitUsings>
     <Nullable>enable</Nullable>
-  </PropertyGroup>
+    <TargetFrameworks>net7.0;net8.0</TargetFrameworks>
 
-  <!-- Remove after switching to .NET 8 SDK -->
-  <PropertyGroup Condition="$([MSBuild]::IsOSPlatform('OSX'))">
-    <TargetFramework>net6.0</TargetFramework>
-    <PublishAot>false</PublishAot>
-  </PropertyGroup>
-  <PropertyGroup Condition="!$([MSBuild]::IsOSPlatform('OSX'))">
-    <TargetFramework>net7.0</TargetFramework>
-    <PublishAot>true</PublishAot>
+    <!-- AOT not supported for osx-arm64 on .NET 7 -->
+    <PublishAot Condition="!$([MSBuild]::IsOSPlatform('OSX')) or '$(TargetFramework)' == 'net8.0'">true</PublishAot>
```

Comment:
@vaind maybe worth putting a TODO in here. Once we get this branch merged with the net8-trimming branch, I think a better way to enable/disable AOT is simply to check whether we're buliding in the context of the `SentryAot.slnf` solution filter, like we do here:
https://github.com/getsentry/sentry-dotnet/blob/731c6829235022c936e7dd53cdcbfbfec389ba76/src/Directory.Build.props#L38-L41

... and here:
https://github.com/getsentry/sentry-dotnet/blob/731c6829235022c936e7dd53cdcbfbfec389ba76/test/Directory.Build.props#L17-L22

We should be able to do this once in the `Directory.Build.props` file in the samples directory.

Reply:
As I've mentioned in the other PR, I am not sure about setting PublishAot globally, i.e. not only for sample apps but also for libraries. Let's see how that pans out. 

In the meantime, I'll add a TODO here but this PR is still addressing the 4.0.0 branch so can't really change any code here.

```suggestion
	<!-- TODO, consider updating after changes from #2732 come into effect -->
    <PublishAot Condition="!$([MSBuild]::IsOSPlatform('OSX')) or '$(TargetFramework)' == 'net8.0'">true</PublishAot>
```